src/commonTest/kotlin/fr/acinq/lightning/blockchain/electrum/ElectrumMiniWalletTest.kt

```diff
@@ -120,7 +145,85 @@ class ElectrumMiniWalletTest : LightningTestSuite() {
+                Triple("14xb2HATmkBzrHf4CR2hZczEtjYpTh92d2", TxId("71b3dbaca67e9f9189dad3617138c19725ab541ef0b49c05a94913e9f28e3f4e") to 0, 5_000_000.sat),
+                Triple("14xb2HATmkBzrHf4CR2hZczEtjYpTh92d2", TxId("21d2eb195736af2a40d42107e6abd59c97eb6cffd4a5a7a7709e86590ae61987") to 0, 5_000_000.sat),
+                Triple("14xb2HATmkBzrHf4CR2hZczEtjYpTh92d2", TxId("74d681e0e03bafa802c8aa084379aa98d9fcd632ddc2ed9782b586ec87451f20") to 1, 5_000_000.sat),
+                Triple("14xb2HATmkBzrHf4CR2hZczEtjYpTh92d2", TxId("563ea83f9641d37a36f9294d172fdb4fb86c19b0e9cac45e0b27610331138775") to 0, 5_000_000.sat),
+                Triple("14xb2HATmkBzrHf4CR2hZczEtjYpTh92d2", TxId("971af80218684017722429be08548d1f30a2f1f220abc064380cbca5cabf7623") to 1, 5_000_000.sat),
+                Triple("14xb2HATmkBzrHf4CR2hZczEtjYpTh92d2", TxId("b1ec9c44009147f3cee26caba45abec2610c74df9751fad14074119b5314da21") to 0, 5_000_000.sat),
+                Triple("1NHFyu1uJ1UoDjtPjqZ4Et3wNCyMGCJ1qV", TxId("602839d82ac6c9aafd1a20fff5b23e11a99271e7cc238d2e48b352219b2b87ab") to 1, 2_000_000.sat),
+            ),
+            actual = walletState.utxos.map {
+                val txOut = it.previousTx.txOut[it.outputIndex]
+                val address = Bitcoin.addressFromPublicKeyScript(Block.LivenetGenesisBlock.hash, txOut.publicKeyScript.toByteArray()).result!!
+                Triple(address, it.previousTx.txid to it.outputIndex, txOut.amount)
+            }.toSet()
+        )
+
+        wallet.stop()
+        client.stop()
+    }
+
+    @Test
+    fun `multiple addresses with generator`() = runSuspendTest(timeout = 15.seconds) {
+        val client = connectToMainnetServer()
+        val wallet = ElectrumMiniWallet(Block.LivenetGenesisBlock.hash, client, this, logger)
+        wallet.addAddressGenerator {
+            when (it) {
+                0 -> "16MmJT8VqW465GEyckWae547jKVfMB14P8" // has utxos
+                1 -> "15NixkHzN4qW5h2kkmUwpUTf5jCEJNCw9o" // no utxo, but already used
+                2 -> "16hBdohKfkzPnDAA1ne3RPX8jjNhWW3eox" // no utxo, but already used
+                3 -> "14xb2HATmkBzrHf4CR2hZczEtjYpTh92d2" // has utxos
+                4 -> "1NHFyu1uJ1UoDjtPjqZ4Et3wNCyMGCJ1qV" // has utxos
+                5 -> "12yyz3bCKjUoQQPaYbrmzYVM4h7bYa9QYj" // no utxo, but already used
+                6 -> "1NtgeYfAGMwt1vqdJrNTRuPu4Hnqpd4sKX" // fresh unused address
+                else -> error("should not go that far")
+            }
+        }
+
+        val walletState = wallet.walletStateFlow.drop(1).first() // first emitted wallet is empty
+
+        // this has been checked on the blockchain
+        assertEquals(4 + 6 + 1, walletState.utxos.size)
+        assertEquals(72_000_000.sat + 30_000_000.sat + 2_000_000.sat, walletState.totalBalance)
+        assertEquals(11, walletState.utxos.size)
```

Comment:
nit: this is already checked above

Reply:
Done in [6eba0f8](https://github.com/ACINQ/lightning-kmp/pull/584/commits/6eba0f8c05bb90c3a74e6b2daf29cd6770bee56d).